### PR TITLE
Trigger management

### PR DESCRIPTION
**Pull request by abaruni**
this PR is for the fetching of the current configuration and status of a feed trigger

### CODE REVIEW

**Comment by abaruni** [comment]
@csantanapr @jasonpet

**Comment by jthomas** [comment]
Can the READ response be extended to include the `triggersLeft` value? 

Related to: https://github.com/apache/incubator-openwhisk/issues/1398

**Comment by jasonpet** [comment]
Not sure how valid issue #1398 is anymore.  That was created before we set infinite triggers as the default for alarms.  The READ simply reads from the database and the triggers left is not stored in the database.  It is an in-memory count on the provider itself.  We really do not want to be doing database writes on every trigger fire.  We could try to read from a provider endpoint to get this value.  However, this is not trivial now that we have multiple provider instances (as part of our redundancy and load balancing support).  We would have to compute the endpoint url to use (by getting the worker id from the db first).

**Comment by jasonpet** [comment]
@abaruni - the legacy alarms code created some trigger entries in the trigger database with a `_` for the namespace.  It was not until the creation of the __OW_NAMESPACE variable that we always started using the resolved namespace.  In your current implementation the lookup for READ would always use the resolved namespace and would never work for users with older triggers that use `_`.  Please see how we handle DELETE and do the same thing for READ.

**Comment by jthomas** [comment]
@jasonpet Thanks for explaining, I see the reasoning that this requirement is less valid with the infinity max triggers value. 

If the "triggers left" values are only stored in memory, does that mean in the event of a provider crash it's going to lose all that state?

**Comment by jasonpet** [comment]
@jthomas - yes, on a crash we would lose state and the count would start back from the beginning (initial value of maxTriggers).  We are implementing a new alarms action that will fire until an end date (will stop once that date has been reached).  This would be a more reliable option and give the user more control when creating a temporary alarms trigger.

File: provider/lib/utils.js [hunk 1]
```diff
@@ -170,7 +170,7 @@ module.exports = function(
                     var updatedTrigger = existing;
                     var status = {
                         'active': false,
-                        'dateChanged': new Date().toISOString(),
+                        'dateChanged': Date.now(),
```
Comment (csantanapr) [inline provider/lib/utils.js:173]: We want seconds not milliseconds , need to devide 1000 and round

**Review by csantanapr — CHANGES_REQUESTED** [review]

File: action/alarmWebAction.js [hunk 2]
```diff
@@ -72,6 +72,34 @@ function main(params) {
...
+                return getTrigger(db, triggerID);
+            })
+            .then(doc => {
+                var body = {
+                    config: {
+                        name: doc.name,
+                        namespace: doc.namespace,
+                        cron: doc.cron,
+                        payload: doc.payload,
+                        maxTriggers: doc.maxTriggers
```
Comment (csantanapr) [inline action/alarmWebAction.js]: Should we not include maxTriggers if we really want to deprecate them

**Comment by csantanapr** [comment]
Need to handle how to deal when an old trigger has a different format for dateChange,

**Comment by jasonpet** [comment]
Reviewed and tested most recent changes and they look good.

**Review by csantanapr — APPROVED** [review]

**Pull request merged**